go/glide/lib.h

```diff
@@ -0,0 +1,88 @@
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/**
+ * FFI compatible version of the ErrorType enum defined in protobuf.
+ */
+typedef enum ErrorType {
+  ClosingError = 0,
+  RequestError = 1,
+  TimeoutError = 2,
+  ExecAbortError = 3,
+  ConnectionError = 4,
+} ErrorType;
+
+/**
+ * A Redis error.
+ */
+typedef struct RedisErrorFFI {
+  const char *message;
+  enum ErrorType error_type;
+} RedisErrorFFI;
+
+/**
+ * The connection response.
+ *
+ * It contains either a connection or an error. It is represented as a struct instead of an enum for ease of use in the wrapper language.
+ */
+typedef struct ConnectionResponse {
+  const void *conn_ptr;
+  const struct RedisErrorFFI *error;
+} ConnectionResponse;
+
+/**
+ * Success callback that is called when a Redis command succeeds.
+ */
+typedef void (*SuccessCallback)(uintptr_t channel_address, const char *message);
+
+/**
+ * Failure callback that is called when a Redis command fails.
```

Comment:
Please, add a doc describing that `RedisErrorFFI` should be freed afterwards

Reply:
Added.

Reply:
Did you update this file after that? I see it on rs part, but it is not reflected there.